### PR TITLE
add wasm backend for bitwiseAnd API

### PR DESCRIPTION
**Pull request by fengwuyao**
To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

### CODE REVIEW

**Review by Linchenn — APPROVED** [review]
LGTM, thanks!

File: tfjs-backend-wasm/src/cc/kernels/BitwiseAnd.cc [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ===========================================================================*/
+
+#ifdef __EMSCRIPTEN__
+#include <emscripten.h>
+#endif
+#include <xnnpack.h>
```
Comment (chunnienc) [inline tfjs-backend-wasm/src/cc/kernels/BitwiseAnd.cc:18]: remove unused headers

File: tfjs-backend-wasm/src/cc/kernels/BitwiseAnd.cc [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+#include <cstddef>
+
+#include "tfjs-backend-wasm/src/cc/binary.h"
+#include "tfjs-backend-wasm/src/cc/util.h"
+
+namespace {
+template <class T>
```
Comment (chunnienc) [inline tfjs-backend-wasm/src/cc/kernels/BitwiseAnd.cc:26]: `template <typename T>` is preferred go/totw/12#what-do-i-need-to-know-about-templates

(we usually follow the internal language style guides even though it's on github)

File: tfjs-backend-wasm/src/cc/kernels/BitwiseAnd.cc [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+inline T bitwiseAnd(T a, T b) {
```
Comment (chunnienc) [inline tfjs-backend-wasm/src/cc/kernels/BitwiseAnd.cc:27]: `BitwiseAndImpl` go/cstyle#Function_Names

File: tfjs-backend-wasm/src/cc/kernels/BitwiseAnd.cc [hunk 4]
```diff
@@ -0,0 +1,51 @@
+/* Copyright 2023 Google LLC.
```
Comment (chunnienc) [inline tfjs-backend-wasm/src/cc/kernels/BitwiseAnd.cc:1]: Can you add a bazel target following other kernels to https://github.com/tensorflow/tfjs/blob/master/tfjs-backend-wasm/src/cc/BUILD.bazel ?
Although our build rule is somehow broken right now which includes all c files when compiling, maintaining a valid BUILD file will make it easier to fix this issue in the future.

File: tfjs-backend-wasm/src/cc/kernels/BitwiseAnd.cc [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+#ifdef __EMSCRIPTEN__
+EMSCRIPTEN_KEEPALIVE
+#endif
+void BitwiseAnd(const size_t a_id, const size_t* a_shape_ptr,
+                const size_t a_shape_len, const size_t b_id,
+                const size_t* b_shape_ptr, const size_t b_shape_len,
+                const DType dtype, const size_t out_id) {
+  binary_i32(a_id, a_shape_ptr, a_shape_len, b_id, b_shape_ptr, b_shape_len,
+             out_id,
+             bitwiseAnd<int32_t>);  // input numbers are ensured to be int32
```
Comment (chunnienc) [inline tfjs-backend-wasm/src/cc/kernels/BitwiseAnd.cc:46]: nit: Sometime it's hard to tell the int32 check is done by which step in the op execution, you know the path span across op function (core), kernel function (backend), and C function (backend wasm internal). I would suggest to put more information here

I usually put comments in this format to make things clear for future reference and maintance: https://github.com/tensorflow/tfjs/blob/master/tfjs-backend-wasm/src/cc/kernels/DenseBincount.cc#L55

File: tfjs-backend-wasm/src/setup_test.ts [hunk 6]
```diff
@@ -396,6 +390,7 @@ const TEST_FILTERS: TestFilter[] = [
   {include: 'asinh '},
   {include: 'diag '},
   {include: 'denseBincount '},
+  {include: 'bitwiseAnd'},
```
Comment (chunnienc) [inline tfjs-backend-wasm/src/setup_test.ts:393]: Now are you sure that the test is actually running? A quick way I usually do is intentionally making a wrong implementation (like `return a & b` -> `return a ^ b`) and see if the tests can fail. If so I can confidently tell that the tests are working.

**Review by chunnienc — CHANGES_REQUESTED** [review]

File: tfjs-backend-wasm/src/kernels/BitwiseAnd.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
+
```
Comment (chunnienc) [inline tfjs-backend-wasm/src/kernels/BitwiseAnd.ts:1]: remove empty first line

File: tfjs-backend-wasm/src/setup_test.ts (hunk 6 above)
Comment (fengwuyao) [reply tfjs-backend-wasm/src/setup_test.ts:393]: Thanks, the test is actually running.

**Review by chunnienc — APPROVED** [review]

**Pull request merged**